HermaFx.DataAnnotations.Tests/DistinctElementsAttributeTest.cs

```diff
@@ -0,0 +1,84 @@
+﻿
+using System.Collections.Generic;
+using NUnit.Framework;
+
+namespace HermaFx.DataAnnotations
+{
+	[TestFixture]
+	public class DistinctElementsAttributeTest
+	{
+
+		private class Model
+		{
+			#region Innter Type
+			public class _Item
+			{
+				public string Property1 { get; set; }
+				public string Property2 { get; set; }
+
+			}
+			#endregion
+
+			[DistinctElements("Property1")]
+			public IEnumerable<_Item> Values { get; set; }
+
+		}
+
+		[Test]
+		public void IsValidTest()
+		{
+			var model = new Model();
+
+			model.Values = new List<Model._Item>()
+			{
+					new Model._Item() { Property1 = "hello", Property2 = "world" },
+					new Model._Item() { Property1 = "hi", Property2 = "world" },
+			};
+
+			Assert.That(ExtendedValidator.IsValid(model), Is.True);
```

Comment:
Please, add some asserts again Validation ensuring thrown exception and/or ValidationResult points to correct named property / element duplicated. Also ensure:

1. Correct (default) message is fired along exception/result.
2. Correct (overriden) message is fired along exception/result when overriding message (thru attribute's ErrorMessage & ErrorMessageResource properties)

Reply:
Done. Is it Ok?

Reply:
Actually I think you current changes test ErrorMessage, but break ErrorMessageResourceName.. :?